test/e2e/node/taints.go

```diff
@@ -451,19 +450,23 @@ var _ = SIGDescribe("NoExecuteTaintManager Multiple Pods [Serial]", func() {
 		framework.ExpectNodeHasTaint(cs, nodeName, &testTaint)
 		defer e2enode.RemoveTaintOffNode(cs, nodeName, testTaint)
 
-		// 3. Wait to see if both pods get evicted in between [5, 25] seconds
-		ginkgo.By("Waiting for Pod1 and Pod2 to be deleted")
-		timeoutChannel := time.NewTimer(time.Duration(kubeletPodDeletionDelaySeconds+3*additionalWaitPerDeleteSeconds) * time.Second).C
-		var evicted int
-		for evicted != 2 {
-			select {
-			case <-timeoutChannel:
-				framework.Failf("Failed to evict all Pods. %d pod(s) is not evicted.", 2-evicted)
-				return
-			case podName := <-observedDeletions:
-				framework.Logf("Noticed Pod %q gets evicted.", podName)
-				evicted++
+		wait.PollImmediate(1*time.Second, (kubeletPodDeletionDelaySeconds+14*additionalWaitPerDeleteSeconds)*time.Second, func() (bool, error) {
```

Comment:
@sjenning why is this not upstream?

Reply:
i'm not aware of any carries we have against the node e2e

Reply:
Redirecting the question to @damemi, the author of the the commit in question (`UPSTREAM: <carry>: Change timeout in eviction e2e test to account for new kubelet delay in pod deletion`).